modules/openid/cms_openid_email_service_parameters.e

```diff
@@ -0,0 +1,260 @@
+		local
+			utf: UTF_CONVERTER
+			l_site_name: READABLE_STRING_8
+			s: detachable READABLE_STRING_32
+			l_contact_email, l_subject_register, l_subject_activate, l_subject_password, l_subject_oauth: detachable READABLE_STRING_8
+		do
+			setup := a_cms_api.setup
+				-- Use global smtp setting if any, otherwise "localhost"
+			smtp_server := utf.escaped_utf_32_string_to_utf_8_string_8 (a_cms_api.setup.text_item_or_default ("smtp", "localhost"))
+			l_site_name := utf.escaped_utf_32_string_to_utf_8_string_8 (a_cms_api.setup.site_name)
+			admin_email := a_cms_api.setup.site_email
+
+			if not admin_email.has ('<') then
+				admin_email := l_site_name + " <" + admin_email +">"
+			end
+
+			if attached {CONFIG_READER} a_cms_api.module_configuration_by_name ({CMS_AUTHENTICATION_MODULE}.name, Void) as cfg then
+				if attached cfg.text_item ("smtp") as l_smtp then
+						-- Overwrite global smtp setting if any.
+					smtp_server := utf.utf_32_string_to_utf_8_string_8 (l_smtp)
+				end
+				s := cfg.text_item ("email")
+				if s /= Void then
+					l_contact_email := utf.utf_32_string_to_utf_8_string_8 (s)
+				end
+				s := cfg.text_item ("subject_register")
+				if s /= Void then
+					l_subject_register := utf.utf_32_string_to_utf_8_string_8 (s)
+				end
+				s := cfg.text_item ("subject_activate")
+				if s /= Void then
+					l_subject_register := utf.utf_32_string_to_utf_8_string_8 (s)
+				end
+				s := cfg.text_item ("subject_password")
+				if s /= Void then
+					l_subject_register := utf.utf_32_string_to_utf_8_string_8 (s)
+				end
+				s := cfg.text_item ("subject_oauth")
+				if s /= Void then
+					l_subject_oauth := utf.utf_32_string_to_utf_8_string_8 (s)
+				end
+
+			end
+			if l_contact_email /= Void then
+				if not l_contact_email.has ('<') then
+					l_contact_email := l_site_name + " <" + l_contact_email + ">"
+				end
+				contact_email := l_contact_email
+			else
+				contact_email := admin_email
+			end
+			if l_subject_register /= Void then
+				contact_subject_register := l_subject_register
+			else
+				contact_subject_register := "Thank you for registering with us."
+			end
+
+			if l_subject_activate /= Void then
+				contact_subject_activate := l_subject_activate
+			else
+				contact_subject_activate := "New account activation token."
+			end
+			if l_subject_password /= Void then
+				contact_subject_password := l_subject_password
+			else
+				contact_subject_password := "Password Recovery."
+			end
+			if l_subject_oauth /= Void then
+				contact_subject_oauth := l_subject_oauth
+			else
+				contact_subject_oauth := "Welcome."
+			end
+
+		end
+
+feature	-- Access
+
+	smtp_server: IMMUTABLE_STRING_8
+
+	admin_email: IMMUTABLE_STRING_8
+
+	contact_email: IMMUTABLE_STRING_8
+			-- Contact email.
+
+	contact_subject_register: IMMUTABLE_STRING_8
+	contact_subject_activate: IMMUTABLE_STRING_8
+	contact_subject_password: IMMUTABLE_STRING_8
+	contact_subject_oauth: IMMUTABLE_STRING_8
+
+	account_activation: STRING
```

Comment:
I am wondering, if those are really needed for openid module.
For me, re_activation and so on are mostly for main auth module .. that manage the real CMS account.